runtime/src/iree/base/string_builder.h

```diff
@@ -131,6 +131,46 @@ IREE_API_EXPORT IREE_PRINTF_ATTRIBUTE(2, 3) iree_status_t
     iree_string_builder_append_format(iree_string_builder_t* builder,
                                       const char* format, ...);
 
+// Lightweight builder for lists of iree_string_pair_t.
+// Includes a side pool for keeping dynamically allocated strings, since it is
+// common to need to create backed, temporary strings when constructing lists
+// of views.
+typedef struct iree_string_pair_builder_t {
```

Comment:
add an accessor for the pair list or something, instead of requiring users to poke the fields
(`iree_string_pair_list_t` ala `iree_string_view_list_t`, then `iree_string_pair_builder_pairs()`)